zrml/neo-swaps/src/traits/liquidity_shares_manager.rs

```diff
@@ -21,8 +21,14 @@ use sp_runtime::{DispatchError, DispatchResult};
 /// Trait for managing pool share tokens and distributing fees to LPs according to their share of
 /// the total issuance of pool share tokens.
 pub trait LiquiditySharesManager<T: Config> {
+    type JoinBenchmarkInfo;
```

Comment:
You need a new enum for each function since they might have different execution paths. I don't know if it's too much. The work lies in writing the benchmarks, not in defining what the possible execution paths are.